src/globus_sdk/services/transfer/client.py

```diff
@@ -2699,3 +2699,199 @@ def endpoint_manager_delete_pause_rule(
             f"/v0.10/endpoint_manager/pause_rule/{pause_rule_id}",
             query_params=query_params,
         )
+
+    # Tunnel methods
+
+    def create_tunnel(
+        self,
+        data: dict[str, t.Any] | CreateTunnelData,
+        *,
+        query_params: dict[str, t.Any] | None = None,
+    ) -> response.GlobusHTTPResponse:
+        """
+        :param data: Parameters for the tunnel creation
+        :param query_params: Any additional parameters will be passed through
+            as query params.
+
+        .. tab-set::
+
+            .. tab-item:: Example Usage
+
+                .. code-block:: python
+
+                    tc = globus_sdk.TunnelClient(...)
+                    result = tc.create_tunnel(data)
+                    print(result["data"]["id"])
+
+            .. tab-item:: API Info
+
+                ``POST /tunnels``
+        """
+        log.debug("TransferClient.create_tunnel(...)")
+        if (
+            "submission_id" not in data["data"]["attributes"]
+            or data["data"]["attributes"]["submission_id"] is MISSING
+        ):
+            log.debug("create_tunnel autofetching submission_id")
+            data["data"]["attributes"]["submission_id"] = self.get_submission_id()[
+                "value"
+            ]
```

Comment:
I think the SDK should emit a UsageError, not a bare KeyError, if someone calls `create_tunnel({})`.

cc @derek-globus , @kurtmckee: I trust both of you to check me on this. Should we add new helpers for this? Or handle it inline here? I'm inclined towards inline handling until we see this occur again, even though we expect JSON:API to produce more of these nested data shapes.